src/install-node.ts

```diff
@@ -28,12 +40,27 @@ export async function installNode(
 	if (!res.ok) {
 		throw new Error(`HTTP request failed: ${res.status}`);
 	}
-	return new Promise((resolve, reject) => {
+	if (platform === 'win32') {
+		// Put it in the `bin` dir for consistency with the tarballs
+		const finalDest = join(dest, 'bin');
+		const zipName = basename(tarballUrl);
+		const zipPath = join(dest, zipName);
+
+		debug('Saving Node.js %s zip file to %o', version, zipPath);
+		await pipe(
+			res.body,
+			createWriteStream(zipPath)
+		);
+
+		debug('Extracting Node.js %s zip file to %o', version, finalDest);
+		const zipFile = await zipFromFile(zipPath);
+		await unzip(zipFile, finalDest, { strip: 1 });
+	} else {
 		debug('Extracting Node.js %s tarball to %o', version, dest);
-		res.body
-			.pipe(createGunzip())
-			.pipe(extract({ strip: 1, C: dest }))
-			.on('error', reject)
-			.on('end', resolve);
-	});
+		await pipe(
+			res.body,
+			createGunzip(),
+			extract({ strip: 1, C: dest })
```

Comment:
Why did you remove `.on('error', reject)`?

Reply:
I guess you removed the Promise to so this will throw on error?

Reply:
It wasn't using `promisepipe()` before. That module catches stream error events and throws them.